helm/helm_template_test.go

```
@@ -63,7 +63,153 @@ func TestRedisSentinelModeRendersRedisAuthAndTLSEnvVars(t *testing.T) {
 
-func deploymentEnvByName(t *testing.T, manifest []byte) map[string]map[string]any {
+// A control-plane deploy upgrading from v2 still needs the flat single-app
+// vars: the legacy import reads them to migrate the app into the database and
+// EXPO_APP_ID keeps serving v1 clients that send no expo-app-id header. They
+// must therefore render (as optional) even when controlPlane=true.
+func TestControlPlaneModeStillRendersLegacyFlatEnvVars(t *testing.T) {
+	if _, err := exec.LookPath("helm"); err != nil {
+		t.Skip("helm is not installed")
+	}
+
+	cmd := exec.Command(
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)